sky/cli.py

```diff
@@ -1002,10 +1003,13 @@ def _make_task_from_entrypoint_with_overrides(
         task.name = name
     task.set_envs(env)
     # TODO(wei-lin): move this validation into Python API.
-    if new_resources.accelerators is not None:
-        acc, _ = list(new_resources.accelerators.items())[0]
-        if acc.startswith('tpu-') and task.num_nodes > 1:
-            raise ValueError('Multi-node TPU cluster is not supported. '
+    if task.num_nodes > 1 and tpu_utils.is_tpu(new_resources):
+        if not tpu_utils.is_tpu_vm(new_resources):
+            raise ValueError(
+                'For TPU Node, multi-node cluster is not supported. '
+                f'Got num_nodes={task.num_nodes}.')
+        if tpu_utils.is_tpu_vm_pod(new_resources):
+            raise ValueError('Multi-node TPU Pod cluster is not supported. '
```

Comment:
Don't we want to enclose this with `ux_utils.print_exception_no_traceback`?

Reply:
Great point. fixed!